autorest/sender.go

```diff
@@ -295,14 +295,23 @@ func doRetryForStatusCodesImpl(s Sender, r *http.Request, count429 bool, attempt
 			return resp, err
 		}
 		delayed := DelayWithRetryAfter(resp, r.Context().Done())
-		if !delayed && !DelayForBackoffWithCap(backoff, cap, attempt, r.Context().Done()) {
+		// enforce a 2 minute cap between requests when 429 status codes are
+		// not going to be counted as an attempt and when the cap is 0.
+		// this should only happen in the absense of a retry-after header.
+		if !count429 && cap == 0 {
```

Comment:
Would this be more readable and also do less in the delayed case if you had a !delayed case and then did this in there?

Reply:
I had thought about that, I think the amount of work done in either case is negligible.  I omitted it as I didn't want to make the condition too complicated.  Not against adding it thought if we think it's an improvement.